exchange/bitswap/decision/engine.go

```diff
@@ -180,20 +180,19 @@ func (e *Engine) MessageReceived(p peer.ID, m bsmsg.BitSwapMessage) error {
 			log.Debug("wants", entry.Key, entry.Priority)
 			l.Wants(entry.Key, entry.Priority)
 			if exists, err := e.bs.Has(entry.Key); err == nil && exists {
-				newWorkExists = true
 				e.peerRequestQueue.Push(entry.Entry, p)
+				newWorkExists = true
```

Comment:
There isn't. I switched the order because I mistakenly believed that assigning to the variable would magically signal to the other goroutine immediately. In the moment, I had forgotten how to write software.